lib/restbase.js

```diff
@@ -33,12 +33,13 @@ function cloneRequest(req) {
     };
 }
 
-function RESTBase(options, req) {
+function RESTBase(options, req, parOptions) {
```

Comment:
I have an improvement in mind, but it's a more major refactoring, so we can deal with it as a followup. Instead of calling `request` on a parent `RESTBase` instance and making it create a chile internally, we could pull the child creation process to the server and call the `request` method directly on a child. I think that might me cleaner, as right now inside RESTBase we have access both to the parent and a child, so it's messy and hard to reason about. 

Some applies the the cloned request.

Anyway, this is just an idea to discuss, no need to block the PR on that.